### PR TITLE
Publish lagom 1.4 modules

### PR DESCRIPTION
**Pull request by fsat**
Also fix broken unit test for the `lagom14ScalaServiceLocatorDns` module.

### CODE REVIEW

**Comment by fsat** [comment]
Running `sbt publishLocal` now publishes the lagom 1.4 modules as well

```
[info] 	published lagom14-scala-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-scala-service-locator-dns_2.11.pom
[info] 	published lagom14-scala-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-scala-service-locator-dns_2.11.jar
[info] 	published lagom14-scala-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-scala-service-locator-dns_2.11-sources.jar
[info] 	published lagom14-scala-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-scala-service-locator-dns_2.11-javadoc.jar
[info] 	published ivy to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml

...

[info] 	published lagom14-java-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-java-service-locator-dns_2.11.pom
[info] 	published lagom14-java-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-java-service-locator-dns_2.11.jar
[info] 	published lagom14-java-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-java-service-locator-dns_2.11-sources.jar
[info] 	published lagom14-java-service-locator-dns_2.11 to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-java-service-locator-dns_2.11-javadoc.jar
[info] 	published ivy to /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml
```

Published artefacts are shown as below:

```
Felixs-MacBook-Pro-2:service-locator-dns felixsatyaputra$ find /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11 /Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11 -type f
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-java-service-locator-dns_2.11-javadoc.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-java-service-locator-dns_2.11-javadoc.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-java-service-locator-dns_2.11-javadoc.jar.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-java-service-locator-dns_2.11.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-java-service-locator-dns_2.11.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-java-service-locator-dns_2.11.jar.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-java-service-locator-dns_2.11.pom
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-java-service-locator-dns_2.11.pom.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-java-service-locator-dns_2.11.pom.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-java-service-locator-dns_2.11-sources.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-java-service-locator-dns_2.11-sources.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-java-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-java-service-locator-dns_2.11-sources.jar.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-scala-service-locator-dns_2.11-javadoc.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-scala-service-locator-dns_2.11-javadoc.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/docs/lagom14-scala-service-locator-dns_2.11-javadoc.jar.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/ivys/ivy.xml.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-scala-service-locator-dns_2.11.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-scala-service-locator-dns_2.11.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/jars/lagom14-scala-service-locator-dns_2.11.jar.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-scala-service-locator-dns_2.11.pom
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-scala-service-locator-dns_2.11.pom.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/poms/lagom14-scala-service-locator-dns_2.11.pom.sha1
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-scala-service-locator-dns_2.11-sources.jar
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-scala-service-locator-dns_2.11-sources.jar.md5
/Users/felixsatyaputra/.ivy2/local/com.lightbend/lagom14-scala-service-locator-dns_2.11/2.1.1-SNAPSHOT/srcs/lagom14-scala-service-locator-dns_2.11-sources.jar.sha1
```

**Comment by TimMoore** [comment]
@fsat there is a build failure:

```
[error] /home/travis/build/typesafehub/service-locator-dns/lagom14-scala-service-locator-dns/src/test/scala/com/lightbend/lagom/scaladsl/dns/DnsServiceLocatorSpec.scala:31: object creation impossible, since method httpFilters in trait BuiltInComponents of type => Seq[play.api.mvc.EssentialFilter] is not defined
[error]   val app = new BuiltInComponentsFromContext(ApplicationLoader.createContext(Environment.simple())) with AhcWSComponents with DnsServiceLocatorComponents {
[error]                 ^
[error] one error found
```

**Comment by fsat** [comment]
@TimMoore - I will take a look. This is quite strange - I did try to build locally and I didn't get this error. Let me try again.

**Comment by TimMoore** [comment]
https://github.com/typesafehub/service-locator-dns/blob/master/lagom14-scala-service-locator-dns/src/test/scala/com/lightbend/lagom/scaladsl/dns/DnsServiceLocatorSpec.scala#L31

Do you know why it doesn't use `LagomApplication` here?

**Comment by fsat** [comment]
@TimMoore - to be honest, I'm not sure why it doesn't use `LagomApplication`. I have now modified `DnsServiceLocatorSpec` to use `LagomApplication`, and the test still works as expected.
